saiversion.h

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (c) 2014 Microsoft Open Technologies, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *    not use this file except in compliance with the License. You may obtain
+ *    a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR
+ *    CONDITIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT
+ *    LIMITATION ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS
+ *    FOR A PARTICULAR PURPOSE, MERCHANTABILITY OR NON-INFRINGEMENT.
+ *
+ *    See the Apache Version 2.0 License for specific language governing
+ *    permissions and limitations under the License.
+ *
+ *    Microsoft would like to thank the following companies for their review and
+ *    assistance with these files: Intel Corporation, Mellanox Technologies Ltd,
+ *    Dell Products, L.P., Facebook, Inc., Marvell International Ltd.
+ *
+ * @file    saiversion.h
+ *
+ * @brief   Define the current version
+ */
+
+#if !defined (__SAIVERSION_H_)
+#define __SAIVERSION_H_
+
+#define SAI_MAJOR 1
+#define SAI_MINOR 7
+#define SAI_REVISION 1
+
+#define SAI_VERSION(maj, min, rev) (100 * (maj) + 10 * (min) + (rev))
+
+#define SAI_API_VERSION SAI_VERSION(SAI_MAJOR, SAI_MINOR, SAI_REVISION)
```

Comment:
can we get a version comparing macro for user to use?

Reply:
In my opinion, it is simple enough to use > < == for comparison in code as described in the doc. I can add more if required.
Did you have an API name in mind? SAI_API_VERSION_GREATER/LESSER?